public/model.js

```diff
@@ -116,7 +110,138 @@ window.model = (() => {
         }
         return model;
     };
-    return main;
+
+    const classes = {};
+
+    classes.user = class User {
+        constructor(data) {
+            // fields from Zulip's API
+            this.avatar_url = data.avatar_url;
+            this.avatar_version = data.avatar_version;
+            this.date_joined = data.date_joined;
+            this.email = data.email;
+            this.full_name = data.full_name;
+            this.is_active = data.is_active;
+            this.is_admin = data.is_admin;
+            this.is_bot = data.is_bot;
+            this.is_guest = data.is_guest;
+            this.is_owner = data.is_owner;
+            this.timezone = data.timezone;
+            this.user_id = data.user_id;
+        }
+
+        get name() {
+            return this.full_name;
+        }
+        set name(name) {
+            this.full_name = name;
+        }
+        get id() {
+            return this.user_id;
+        }
+        set id(id) {
+            this.user_id = id;
+        }
+    };
+
+    classes.stream = class Stream {
+        constructor(data) {
+            // fields from Zulip's API
+            this.name = data.name;
+            this.stream_id = data.stream_id;
+            this.date_created = data.date_created;
+            this.description = data.description;
+            this.first_message_id = data.first_message_id;
+            this.history_public_to_subscribers =
+                data.history_public_to_subscribers;
+            this.invite_only = data.invite_only;
+            this.is_announcement_only = data.is_announcement_only;
+            this.is_web_public = data.is_web_public;
+            this.message_retention_days = data.message_retention_days;
+            this.rendered_description = data.rendered_description;
+            this.stream_post_policy = data.stream_post_policy;
+            this.subscribers = data.subscribers;
+        }
+
+        get id() {
+            return this.stream_id;
+        }
+        set id(id) {
+            this.stream_id = id;
+        }
+
+        get subs() {
```

Comment:
This right here is something I want to optimize later. We could just store these at time of object creation if we can

- guarantee that Users will always be loaded before Streams.
- ensure that updates to `this.subscribers` (if someone subs/unsubs) are mirrored here.

The second point here felt like too much code to write and we're just hitting nightly so far with its 50 or so users; we can delay optimizations and just get working code first.